Service/SamlDTO.php

```diff
@@ -118,14 +120,161 @@ public function __construct(array $samlAttributes)
             }
-        });
-
-        /* @var int[] $ids */
-        $this->companyId = $ids[0]; // @phpstan-ignore assign.propertyType
-        $this->divisionId = $ids[1]; // @phpstan-ignore assign.propertyType
-        $this->departmentId = $ids[2]; // @phpstan-ignore assign.propertyType
-        $this->subDepartmentId = $ids[3]; // @phpstan-ignore assign.propertyType
-        $this->officeId = $ids[4]; // @phpstan-ignore assign.propertyType
+
+            $departmentIds[] = $value;
+        }
+
+        $this->departmentIds = $departmentIds;
+
+        if (0 === count($this->departmentIds)) {
+            throw new AakSamlException(sprintf('No organization ids given in claims: "%s"', $this->getAttributeValue(self::ID_ARRAY_ATTRIBUTE, $samlAttributes)));
+        }
+
+        // We can map claims from level 1-5. If we see id's for deeper levels we don't know the corresponding claim for
+        // department name.
+        $this->companyId = array_key_exists(0, $this->departmentIds) ? $this->departmentIds[0] : null;
+        $this->divisionId = array_key_exists(1, $this->departmentIds) ? $this->departmentIds[1] : null;
+        $this->departmentId = array_key_exists(2, $this->departmentIds) ? $this->departmentIds[2] : null;
+        $this->subDepartmentId = array_key_exists(3, $this->departmentIds) ? $this->departmentIds[3] : null;
+        $this->officeId = array_key_exists(4, $this->departmentIds) ? $this->departmentIds[4] : null;
+    }
+
+    /**
+     * Is the user a team lead.
+     *
+     * @return bool
+     */
+    public function isTeamLead(): bool
+    {
+        // If claims list the user as their own manager they are team lead for the lowest org unit given in the id's
+        return $this->emailAddress === $this->managerEmail;
+    }
+
+    /**
+     * Get the organization id. This will be the last of the ids given in the claims.
+     *
+     * @return int
+     */
+    public function getOrganizationUnitId(): int
+    {
+        $id = array_slice($this->departmentIds, -1, 1);
+
+        return $id[0];
+    }
+
+    /**
+     * Get the organization name the user is placed in. For levels 1-5 we know the claim to map to. For deeper levels
+     * the claims are unknown.
+     *
+     * @return string
+     */
+    public function getOrganizationUnitName(): string
+    {
+        $depth = \count($this->departmentIds);
+
+        return $this->getOrgUnitName($depth);
+    }
+
+    /**
+     * Get the organization id the user is a member in. This will be the last of the ids given in the claims for
+     * employees and the second to last for team leads.
+     *
+     * @return int
+     *
+     * @throws AakSamlException
+     */
+    public function getMemberOrganizationUnitId(): int
+    {
+        if ($this->isTeamLead()) {
+            $id = array_slice($this->departmentIds, -2, 1);
+        } else {
+            $id = array_slice($this->departmentIds, -1, 1);
```

Comment:
A private `getDepatmentId($offsetFromEnd)` function may come in handy for reducing code duplication of the array slicing (and throwing exceptions).